src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs

```diff
@@ -54,10 +56,23 @@ public bool IsBrokerInstalledAndInvokable()
             }
         }
 
+        private void CheckPowerOptimizationStatus(bool CheckPowerOptimizations = true)
+        {
+            var powerManager = PowerManager.FromContext(Application.Context);
+            if (powerManager.IsDeviceIdleMode &&
+                !powerManager.IsIgnoringBatteryOptimizations(Application.Context.PackageName) )
+            {
+                _logger.Warning("Power optimization detected on the client application: " + Application.Context.PackageName + "\nPlease disable power optimizations.");
+                throw new MsalClientException(MsalError.AndroidBrokerClientPowerOptimization);
```

Comment:
Please throw exceptions that have both error code and error message.

Reply:
we should throw the same exception + error message as the android team, i think.

---
In reply to: [533575764](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2257#discussion_r533575764) [](ancestors = 533575764)